### PR TITLE
Replace project path with directory

### PR DESCRIPTION
**Pull request by davkean**
Fixes: #2551

This was allocating up to 5.5% of allocations:

image

### CODE REVIEW

File: src/Build/Evaluation/ItemSpec.cs [hunk 1]
```diff
@@ -300,22 +297,22 @@ public virtual IMSBuildGlob ToMSBuildGlob()
 
         protected virtual IMSBuildGlob CreateMsBuildGlob()
         {
-            return Globbing.MSBuildGlob.Parse(ProjectPath, ItemSpecFragment.Unescape());
+            return Globbing.MSBuildGlob.Parse(ProjectDirectory, ItemSpecFragment.Unescape());
```
Comment (davkean) [inline src/Build/Evaluation/ItemSpec.cs]: @cdmihai Can you help me understand this? This was passing the project file name - but it appears that MSBuildGlob expects a "root" path. How is this supposed to work?

**Comment by davkean** [comment]
Tests are failing right now - need to figure out why.

**Comment by davkean** [comment]
@AndyGerlicher @cdmihai This is ready for review.

File: src/Build/Evaluation/ItemSpec.cs (hunk 1 above)
Comment (davkean) [reply src/Build/Evaluation/ItemSpec.cs]: We talked offline, it works accidentally.

**Comment by davkean** [comment]
tag @pilchie

**Comment by davkean** [comment]
@dotnet-bot retest Windows_NT Build for Full please

**Comment by AndyGerlicher** [comment]
Nice!! This this table might be getting a little out of hand though...

### Evaluation: Time (ms)
Test | Overall | Significant δ  | Value
:--- | :--- | :--- | :---
DotnetConsoleProject | :white_check_mark: | yes | 30.6228 -> 30.3252 (-0.972%)
DotnetWebProject | ::ok_hand: | no | 49.9704 -> 49.9008 (-0.139%)
DotnetMvcProject | :white_check_mark: | yes | 87.7896 -> 86.3275 (-1.665%)
Picasso | :white_check_mark: | yes | 266.3487 -> 258.0688 (-3.109%)
SmallP2POldCsproj | :white_check_mark: | yes | 42.6873 -> 42.1706 (-1.21%)
SmallP2PNewCsproj | :white_check_mark: | yes | 84.3846 -> 82.1966 (-2.593%)
Generated_100_100_v150 | :white_check_mark: | yes | 1324.1047 -> 1245.2495 (-5.955%)
LargeP2POldCsproj | :white_check_mark: | yes | 764.7627 -> 753.0463 (-1.532%)
roslyn | :white_check_mark: | yes | 5427.0552 -> 5332.4751 (-1.743%)

### Evaluation: Memory (bytes)
Test | Overall | Significant δ  | Value
:--- | :--- | :--- | :---
DotnetConsoleProject | :white_check_mark: | yes | 4985623 -> 4935865 (-0.998%)
DotnetWebProject | :white_check_mark: | yes | 6895078 -> 6837016 (-0.842%)
DotnetMvcProject | :white_check_mark: | yes | 9139645 -> 9016288 (-1.35%)
Picasso | :white_check_mark: | yes | 37068367 -> 36036141 (-2.785%)
SmallP2POldCsproj | :white_check_mark: | yes | 6115583 -> 6009882 (-1.728%)
SmallP2PNewCsproj | :white_check_mark: | yes | 10815779 -> 10586858 (-2.117%)
Generated_100_100_v150 | :white_check_mark: | yes | 155739686 -> 150854971 (-3.136%)
LargeP2POldCsproj | :white_check_mark: | yes | 90066052 -> 87362770 (-3.001%)
roslyn | :white_check_mark: | yes | 598826584 -> 586491058 (-2.06%)

### Build: Time (ms)
Test | Overall | Significant δ  | Value
:--- | :--- | :--- | :---
DotnetConsoleProject | :red_circle: | yes | 7.5941 -> 7.7116 (1.547%)
DotnetWebProject | :red_circle: | yes | 9.4014 -> 9.4516 (0.534%)
DotnetMvcProject | :white_check_mark: | yes | 10.2089 -> 10.1036 (-1.031%)
Picasso | :white_check_mark: | yes | 2368.4776 -> 2329.3031 (-1.654%)
SmallP2POldCsproj | :white_check_mark: | yes | 85.803 -> 85.231 (-0.667%)
SmallP2PNewCsproj | :white_check_mark: | yes | 88.8488 -> 87.8034 (-1.177%)

### Build: Memory (bytes)
Test | Overall | Significant δ  | Value
:--- | :--- | :--- | :---
DotnetConsoleProject | :ok_hand: | no | 1164515 -> 1164592 (0.007%)
DotnetWebProject | :white_check_mark: | yes | 1304021 -> 1302886 (-0.087%)
DotnetMvcProject | :ok_hand: | no | 1480682 -> 1480964 (0.019%)
Picasso | :white_check_mark: | yes | 290188880 -> 278422956 (-4.055%)
SmallP2POldCsproj | :white_check_mark: | yes | 10033071 -> 9932713 (-1%)
SmallP2PNewCsproj | :white_check_mark: | yes | 10212052 -> 10110180 (-0.998%)

**Pull request merged**